File: container/Hero.js
import React, { useState } from "react";
import { Row, Col } from "react-bootstrap";
import Typewriter from "typewriter-effect";
import Glitch from "../components/Glitch";
import ThemeContext from "../utils/context";
import heroData from "../data/hero.json";

function Hero({ id }) {
  const [height, setHeight] = useState(0);
  return (
    <section id={`${id}`} className="hero" style={{ height: "auto" }}>
      <Row>
        <Col md={6} className="content">
          <div className="content-text">
            <div className="line-text">
              <h4>俺は</h4>
            </div>
            <Glitch text="Lucas Goldner" />
            <Typewriter
              options={{
                strings: [
                  "IOS App Developer",
                  "Front End Engineer",
                  "Android App Developer",
                ],
                autoStart: true,
                loop: true,
              }}
            />
            <button className="hover-button">
              <span>Download CV</span>
            </button>
          </div>
          {heroData.icons.edges.map((value, index) => {
            return (
              <img
                src={value.node.childImageSharp.fluid.src}
                className={`animated fadeIn move-${
                  Math.floor(Math.random() * 10) % 2 === 0 ? "up" : "down"
                } float-image`}
                style={{
                  left: `${index * 10}%`,
                  bottom: `${
                    Math.random() *
                      (+(index % 2 === 0 ? 80 : 20) -
                        +(index % 2 === 0 ? 70 : 10)) +
                    +(index % 2 === 0 ? 70 : 10)
                  }%`,
                }}
                alt="shape"
                key={index}
              />
            );
          })}
        </Col>
        <Col md={6} className="img">
          <img src={heroData.Img.childImageSharp.fluid.src} alt="person" />
        </Col>
      </Row>
    </section>
  );
}

export default Hero;

File: lib/sections/hero.dart
import 'package:jaspr/dom.dart';
import 'package:jaspr/jaspr.dart';

import '../components/gde_badge.dart';
import '../components/glitch.dart';
import '../components/hover_button.dart';
import '../components/typewriter.dart';
import '../data/site_data.dart';
import '../interop/browser.dart';
import '../layout/metrics.dart';

/// The landing panel: name, rotating job title, CV button and the photo.
class Hero extends StatelessComponent {
  const Hero({super.key});

  @override
  Component build(BuildContext context) {
    final metrics = MetricsProvider.of(context);

    return section(
      id: 'home',
      classes: 'hero',
      styles: Styles(raw: {'height': metrics.cssHeight}),
      [
        div(classes: 'row', [
          div(classes: 'content col-md-6', [
            div(classes: 'content-text', [
              div(classes: 'line-text', [
                h4([.text('俺は')]),
              ]),
              const Glitch('Lucas Goldner'),
              const Typewriter(strings: typewriterStrings),
              const GdeBadge(),
              HoverButton(
                label: 'Download CV',
                onClick: () => openUrl(cvUrl),
              ),
            ]),
            ..._floatingIcons(),
          ]),
          div(classes: 'img col-md-6', [
            img(src: heroImage, alt: heroImageAlt),
          ]),
        ]),
      ],
    );
  }

  /// The hobby icons bobbing over the dark panel.
  List<Component> _floatingIcons() {
    return [
      for (var i = 0; i < heroIcons.length; i++)
        img(
          src: heroIcons[i].src,
          alt: 'shape',
          classes:
              'animated fadeIn '
              'move-${heroIcons[i].moveUp ? 'up' : 'down'} float-image',
          styles: Styles(
            raw: {
              'left': '${i * 10}%',
              'bottom': '${heroIcons[i].bottomPercent}%',
            },
          ),
        ),
    ];
  }
}
